### ISSUE TITLE
Generate correct config.h depending on arch

### ISSUE THREAD

**Issue by mbrt** [mbrt/librsync-rs#1]
Needs to test against [TARGET](http://doc.crates.io/environment-variables.html#environment-variables-cargo-sets-for-build-scripts) env var in the build script, to generate the correct defines, instead of using a pre-generated `config.h`.

- [x] Generate config files in officially supported Rust distributions;
- [x] Make a rust binary that generates them automatically, by using cmake crate.

**Comment by mbrt** [comment]
Here is the generated config from VS2015 x86: [config-msvc-x86.zip](https://github.com/mbrt/librsync-rs/files/94150/config-msvc-x86.zip).

**Comment by mbrt** [comment]
DIfference between `linux-x86_64` and `linux-i686`:
```
#define SIZEOF_LONG 4
#define SIZEOF_SIZE_T 4
#define SIZEOF_UNSIGNED_LONG 4
```

**Comment by mbrt** [comment]
Fixed in 6c9d007.
